stdlib/InteractiveUtils/src/diagnostics.jl

```diff
@@ -0,0 +1,122 @@
+"""
+    InteractiveUtils.diagnostics(io::IO=stderr)
+
+Print a variety of useful debugging info.
+
+!!! warning "Warning"
+    The output of this function may contain sensitive information. Before sharing the output,
+    please review the output and remove any data that should not be shared publicly.
+
+See also: [`versioninfo`](@ref), [`stdlib_diagnostics`](@ref).
+"""
+diagnostics()                        = diagnostics(stderr)
+diagnostics(modules::Vector{Module}) = diagnostics(stderr, modules)
+diagnostics(m::Module)               = diagnostics(stderr, m)
+function diagnostics(io::IO)
+    # Get the list of all modules that are currently loaded
+    loaded_modules = copy(Base.loaded_modules_array())
+    sort!(loaded_modules; by = nameof)
+    diagnostics(io, loaded_modules)
+    return nothing
+end
+function diagnostics(io::IO, modules::Vector{Module})
+    for m in modules
+        diagnostics(io, m)
+    end
+    return nothing
+end
+function diagnostics(io::IO, m::Module)
+    module_name = nameof(m)
+    if isdefined(m, :__diagnostics__) && applicable(m.__diagnostics__, io)
+        header = "# $(module_name)"
+        println(io, header)
+        try
+            m.__diagnostics__(io)
+        catch ex
+            bt = catch_backtrace()
+            msg = "Warning: caught an error while running $(module_name).__diagnostics__()"
+            println(io, msg)
+            Base.showerror(io, ex)
+            Base.show_backtrace(io, bt)
```

Comment:
```suggestion
            Base.showerror(io, catch_stack())
```

Reply:
Per https://docs.julialang.org/en/v1.8-dev/base/base/#Base.current_exceptions, it looks like `catch_stack()` has been renamed in 1.7 and later?